account_financial_report/models/account_move_line.py

```diff
@@ -12,6 +12,15 @@ class AccountMoveLine(models.Model):
         "account.analytic.account", compute="_compute_analytic_account_ids", store=True
     )
 
+    analytic_plan_ids = fields.Many2many(
+        "account.analytic.plan", compute="_compute_analytic_plan_ids", store=True
+    )
+
+    @api.depends("analytic_account_ids")
+    def _compute_analytic_plan_ids(self):
+        self.analytic_plan_ids = False
+        self.analytic_plan_ids = self.analytic_account_ids.plan_id
```

Comment:
You have to iterate over the recordset or you'll assign all the plans to al the move lines